### DISCUSSION TITLE
OTP Authentication

### DISCUSSION THREAD

**Discussion by IDJGILL** [AxeWP/wp-graphql-headless-login#103] [General]
Hi, want some help with implementing the OTP login and sign up. I need some tips how can I do that. Thanks by the way for this awsome plugin.

**Comment by IDJGILL** [comment]
> Hi, want some help with implementing the OTP login and sign up. I need some tips how can I do that. Thanks by the way for this awsome plugin.

I think I found the solution by going though your documentation, and I found that we can use Site Token for it. It makes it easy to just send secret header when we verify the user at our end and ask WordPress for auth tokens. I might be wrong but I want to know your thoughts on this. Thanks.

**Reply by justlevine** [reply]
Hey @IDJGILL,

As you noted, you can currently implement OTP by using a site token, just make sure to take care to strictly validate all aspects of the one-time passcode (and of course only use the site token on a server-side request). The site token let's you log in as anyone, it trusts you to secure it's usages.

An explicit OTP implementation is indeed on the roadmap, which would offload these security concerns to the plugin instead of your frontend code.

**Comment by IDJGILL** [comment]
> Hey @IDJGILL,
> 
> As you noted, you can currently implement OTP by using a site token, just make sure to take care to strictly validate all aspects of the one-time passcode (and of course only use the site token on a server-side request). The site token let's you log in as anyone, it trusts you to secure it's usages.
> 
> An explicit OTP implementation is indeed on the roadmap, which would offload these security concerns to the plugin instead of your frontend code.

Thanks for the quick response and yeah until a official OTP implementation comes up I'll be making sure and follow the guidelines as mention in your documentation.
